src/pool/pool_jemalloc.c

```diff
@@ -0,0 +1,397 @@
+// remain distinct mappings and therefore should continue to be operated on independently.
+// (from https://jemalloc.net/jemalloc.3.html)
+static bool arena_extent_merge(extent_hooks_t *extent_hooks, void *addr_a,
+                               size_t size_a, void *addr_b, size_t size_b,
+                               bool committed, unsigned arena_ind) {
+    (void)extent_hooks; // unused
+    (void)addr_a;       // unused
+    (void)addr_b;       // unused
+    (void)size_a;       // unused
+    (void)size_b;       // unused
+    (void)committed;    // unused
+    (void)arena_ind;    // unused
+
+    return true; // true means failure (unsupported)
+}
+
+// The extent_hooks_t structure comprises function pointers which are described individually below.
+// jemalloc uses these functions to manage extent lifetime, which starts off with allocation
+// of mapped committed memory, in the simplest case followed by deallocation. However, there are
+// performance and platform reasons to retain extents for later reuse. Cleanup attempts cascade
+// from deallocation to decommit to forced purging to lazy purging, which gives the extent
+// management functions opportunities to reject the most permanent cleanup operations in favor
+// of less permanent (and often less costly) operations. All operations except allocation can be
+// universally opted out of by setting the hook pointers to NULL, or selectively opted out of
+// by returning failure. Note that once the extent hook is set, the structure is accessed directly
+// by the associated arenas, so it must remain valid for the entire lifetime of the arenas.
+// (from https://jemalloc.net/jemalloc.3.html)
+static extent_hooks_t arena_extent_hooks = {
+    .alloc = arena_extent_alloc,
+    .dalloc = arena_extent_dalloc,
+    .destroy = arena_extent_destroy,
+    .commit = arena_extent_commit,
+    .decommit = arena_extent_decommit,
+    .purge_lazy = arena_extent_purge_lazy,
+    .purge_forced = arena_extent_purge_forced,
+    .split = arena_extent_split,
+    .merge = arena_extent_merge,
+};
+
+static void *je_malloc(void *pool, size_t size) {
+    assert(pool);
+    jemalloc_memory_pool_t *je_pool = (jemalloc_memory_pool_t *)pool;
+    int flags = MALLOCX_ARENA(je_pool->arena_index);
+    void *ptr = mallocx(size, flags);
+    if (ptr == NULL) {
+        TLS_last_allocation_error = UMF_RESULT_ERROR_OUT_OF_HOST_MEMORY;
+        return NULL;
+    }
+
+    return ptr;
+}
+
+static umf_result_t je_free(void *pool, void *ptr) {
+    (void)pool; // unused
+    assert(pool);
+
+    if (ptr != NULL) {
+        dallocx(ptr, 0);
+    }
+
+    return UMF_RESULT_SUCCESS;
+}
+
+static void *je_calloc(void *pool, size_t num, size_t size) {
+    assert(pool);
+    size_t csize = num * size;
+    void *ptr = je_malloc(pool, csize);
+    if (ptr == NULL) {
+        // TLS_last_allocation_error is set by je_malloc()
+        return NULL;
+    }
+
+    memset(ptr, 0, csize);
```

Comment:
The same comment as above that memory can be not accessible. Please add the same TODO comment.

Reply:
TODO added

Reply:
@vinser52 Done